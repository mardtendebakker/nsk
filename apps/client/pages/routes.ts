export const DASHBOARD = '/';
export const SIGN_IN = '/sign-in';
