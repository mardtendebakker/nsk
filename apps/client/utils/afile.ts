import { AFile } from './axios/models/aFile';

export function buildAFileLink(afile: AFile): string {
  return `https://${afile.unique_server_filename}.s3.amazonaws.com/${afile.discr}/${afile.original_client_filename}`;
}
