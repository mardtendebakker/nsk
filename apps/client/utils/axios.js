import axios from 'axios';

export { AxiosPromise, AxiosResponse, AxiosError } from 'axios';

const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3333' : 'production_url';

export const client = axios.create({
  baseURL,
  timeout: 120000,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
});

export const { CancelToken } = axios;

export const SUPLIERS_PATH = '';

export default client;
