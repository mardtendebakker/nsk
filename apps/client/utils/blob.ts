export function openBlob(blob: Blob): void {
  const a = document.createElement('a');
  document.body.appendChild(a);
  a.style.cssText = 'display: none';

  const url = window.URL.createObjectURL(blob);

  a.href = url;
  a.target = '_blank';
  a.click();
  window.URL.revokeObjectURL(url);
}
