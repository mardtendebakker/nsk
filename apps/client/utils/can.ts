import _ from 'lodash';
import { Group } from '../stores/security/types';

export default function can(userGroups: Group[], requiredGroups: Group[], disableDefaultGroups = false): boolean {
  return _.intersection(userGroups, disableDefaultGroups ? requiredGroups : [...requiredGroups, 'admin', 'super_admin']).length > 0;
}
