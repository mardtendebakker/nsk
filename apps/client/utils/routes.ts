export const DASHBOARD = '/';
export const SIGN_IN = '/sign-in';
export const CUSTOMERS_CONTACTS = '/customers/contacts';
export const CUSTOMERS_CONTACTS_NEW = '/customers/contacts/new';
export const CUSTOMERS_EMAILS_NEW = '/customers/emails/new';
export const CUSTOMERS_EMAILS = '/customers/emails';
export const SUPPLIERS = '/suppliers/:id';
export const ACCOUNT_VERIFICATION = '/account-verification';
export const SETTINGS = '/settings';
export const ORDERS_PURCHASES = '/orders/purchases';
export const ORDERS_SALES = '/orders/sales';
export const STOCKS_PRODUCTS = '/stock/products';
export const STOCKS_REPAIR_SERVICES = '/stock/repair-services';
