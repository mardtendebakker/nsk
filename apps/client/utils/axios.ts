import axios from 'axios';

export { AxiosError } from 'axios';
export type { AxiosPromise, AxiosResponse } from 'axios';

const baseURL = process.env.NODE_ENV === 'development' ? 'http://localhost:3333/api' : process.env.NX_AXIOS_BASE_URL;

export const client = axios.create({
  baseURL,
  timeout: 120000,
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
});

export const { CancelToken } = axios;

export const CUSTOMERS_PATH = '/customers/:id';
export const SUPPLIERS_PATH = '/suppliers/:id';

export default client;

export interface Customer {
  id: number,
  name?: string,
  representative?: string,
  kvk_nr?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
  is_partner?: number,
}

export interface Supplier {
  id: number,
  name?: string,
  representative?: string,
  email?: string,
  phone?: string,
  phone2?: string,
  street?: string,
  street_extra?: string,
  city?: string,
  country?: string,
  state?: string,
  zip?: string,
  street2?: string,
  street_extra2?: string,
  city2?: string,
  country2?: string,
  state2?: string,
  zip2?: string,
  partner?: string,
}
