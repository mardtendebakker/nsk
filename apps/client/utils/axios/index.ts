import axios from 'axios';
import axiosClient from './client';
import refreshToken from './refreshTokenInterceptor';

export {
  CUSTOMERS_PATH,
  SUPPLIERS_PATH,
  SIGN_IN_PATH,
  REFRESH_TOKEN_PATH,
  USER_INFO_PATH,
  SIGN_UP_PATH,
  CONFIRM_ACCOUNT_PATH,
  CHANGE_PASSWORD_PATH,
  SEND_VERIFICATION_CODE_PATH,
  FORGOT_PASSWORD_PATH,
  SALES_ORDERS_PATH,
  PURCHASE_ORDERS_PATH,
  STOCK_PRODUCTS_PATH,
  USERS_PATH,
  STOCK_REPAIR_SERVICES_PATH,
  ORDER_STATUSES_PATH,
  COMPANIES_PATH,
  LOCATIONS_PATH,
  BULK_PRINT_SALES_PATH,
  BULK_PRINT_PURCHASES_PATH,
  ORDERS_ANALYTICS_PATH,
} from './paths';
export { AxiosError } from 'axios';
export type { AxiosPromise, AxiosResponse } from 'axios';

axiosClient.interceptors.response.use((response) => response, refreshToken);

export const { CancelToken } = axios;

export default axiosClient;
