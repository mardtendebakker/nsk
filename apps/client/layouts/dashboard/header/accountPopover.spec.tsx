import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import AccountPopover from './accountPopover';
import { ADMIN_USERS, SETTINGS } from '../../../utils/routes';

const mockUseSecurity = {
  signOut: jest.fn(() => Promise.resolve()),
  state: { user: null },
};

jest.mock('../../../hooks/useSecurity', () => jest.fn(() => mockUseSecurity));

const mockRouter = {
  push: jest.fn(),
};

jest.mock('next/router', () => ({
  useRouter: () => mockRouter,
}));

describe('AccountPopover', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<AccountPopover />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should open the popover when the button is clicked', () => {
    const { getByRole, getByText } = render(<AccountPopover />);
    const button = getByRole('button');
    fireEvent.click(button);
    expect(getByText(/settings/i)).toBeInTheDocument();
  });

  it('should not display menu by default', () => {
    const { queryByText } = render(<AccountPopover />);
    queryByText('logout');
    expect(queryByText('logout')).toBeNull();
  });

  it('should logout', () => {
    const { getByRole, getByText } = render(<AccountPopover />);
    const button = getByRole('button');
    fireEvent.click(button);
    const logoutButton = getByText('logout');
    fireEvent.click(logoutButton);
    expect(mockUseSecurity.signOut).toBeCalled();
  });

  it('should navigate to admin', () => {
    const { getByRole, getByText } = render(<AccountPopover />);
    const button = getByRole('button');
    fireEvent.click(button);
    const logoutButton = getByText('admin');
    fireEvent.click(logoutButton);
    expect(mockRouter.push).toBeCalledWith(ADMIN_USERS);
  });

  it('should navigate to settings', () => {
    const { getByRole, getByText } = render(<AccountPopover />);
    const button = getByRole('button');
    fireEvent.click(button);
    const logoutButton = getByText('settings');
    fireEvent.click(logoutButton);
    expect(mockRouter.push).toBeCalledWith(SETTINGS);
  });
});
