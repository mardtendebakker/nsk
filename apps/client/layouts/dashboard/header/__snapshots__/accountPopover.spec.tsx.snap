// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountPopover matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1dnh70-MuiAvatar-root"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
      data-testid="PersonIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
      />
    </svg>
  </div>
  <h5
    class="MuiTypography-root MuiTypography-h5 css-18ezm2a-MuiTypography-root"
  />
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uym7v6-MuiSvgIcon-root"
      data-testid="ChevronRightIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
